test/select-prompt.test.js

```diff
@@ -0,0 +1,344 @@
+      choices: [
+        { value: 'foo', label: 'foo' },
+        { value: 'bar', label: 'bar' }
+      ]
+    }
+    const inputs = [
+      { name: 'down' },
+      { name: 'down' },
+      { name: 'return' }
+    ]
+    const selectPrompt = await TestingPrompt.SelectPrompt(
+      message,
+      options,
+      inputs,
+      (log) => logs.push(log)
+    )
+
+    const input = await selectPrompt.select()
+
+    assert.deepEqual(logs, [
+      '? Choose between foo & bar',
+      ' › foo',
+      '   bar',
+      '   foo',
+      ' › bar',
+      ' › foo',
+      '   bar',
+      '✔ Choose between foo & bar › foo'
+    ])
+    assert.deepEqual(input, 'foo')
+  })
+
+  it('It should ignore foo.', async () => {
+    const logs = []
+    const message = 'Choose between foo & bar'
+    const options = {
+      choices: [
+        { value: 'foo', label: 'foo' },
+        { value: 'bar', label: 'bar' }
+      ],
+      ignoreValues: ['foo']
+    }
+    const inputs = [{ name: 'return' }]
+    const selectPrompt = await TestingPrompt.SelectPrompt(
+      message,
+      options,
+      inputs,
+      (log) => logs.push(log)
+    )
+
+    const input = await selectPrompt.select()
+
+    assert.deepEqual(logs, [
+      '? Choose between foo & bar',
+      ' › foo',
+      '   bar'
+      // '✔ Choose between foo & bar › foo'
+    ])
+    assert.deepEqual(input, 'foo')
+  })
+
+  it('Should display 5 choices and allow scrolling.', async () => {
+    const logs = []
+    const message = 'Choose option'
+    const options = {
+      choices: [
+        { value: 'option1', label: 'Option 1' },
+        { value: 'option2', label: 'Option 2' },
+        { value: 'option3', label: 'Option 3' },
+        { value: 'option4', label: 'Option 4' },
+        { value: 'option5', label: 'Option 5' },
+        { value: 'option6', label: 'Option 6' },
+        { value: 'option7', label: 'Option 7' },
+        { value: 'option8', label: 'Option 8' },
+        { value: 'option9', label: 'Option 9' },
+        { value: 'option10', label: 'Option 10' }
+      ],
+      maxVisible: 5
+    }
+    const inputs = [
+      { name: 'down' },
+      { name: 'down' },
+      { name: 'down' },
+      { name: 'return' }
+    ]
+    const selectPrompt = await TestingPrompt.SelectPrompt(
+      message,
+      options,
+      inputs,
+      (log) => logs.push(log)
+    )
+
+    const input = await selectPrompt.select()
+
+    assert.deepEqual(logs, [
+      '? Choose option',
+      ' › Option 1 ',
+      '   Option 2 ',
+      '   Option 3 ',
+      '   Option 4 ',
+      '⭣  Option 5 ',
+      '   Option 1 ',
+      ' › Option 2 ',
+      '   Option 3 ',
+      '   Option 4 ',
+      '⭣  Option 5 ',
+      '   Option 1 ',
+      '   Option 2 ',
+      ' › Option 3 ',
+      '   Option 4 ',
+      '⭣  Option 5 ',
+      '⭡  Option 2 ',
+      '   Option 3 ',
+      ' › Option 4 ',
+      '   Option 5 ',
+      '⭣  Option 6 ',
+      '✔ Choose option › Option 4'
```

Comment:
I confess that it's a bit hard to understand this string; I think that it's due to the scrolling rule.

Could you add comment to explain it?